hazelcast-jet-hadoop/src/main/java/com/hazelcast/jet/connector/hadoop/WritableSerializerHooks.java

```diff
@@ -0,0 +1,157 @@
+import com.hazelcast.nio.serialization.SerializerHook;
+import com.hazelcast.nio.serialization.StreamSerializer;
+import org.apache.hadoop.io.BooleanWritable;
+import org.apache.hadoop.io.ByteWritable;
+import org.apache.hadoop.io.DoubleWritable;
+import org.apache.hadoop.io.FloatWritable;
+import org.apache.hadoop.io.IntWritable;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.Writable;
+
+import java.io.IOException;
+
+import static com.hazelcast.jet.impl.util.ExceptionUtil.rethrow;
+
+final class WritableSerializerHooks {
+
+    private static final int DEFAULT = -380;
+    private static final int INT = -381;
+    private static final int LONG = -382;
+    private static final int BOOLEAN = -383;
+    private static final int FLOAT = -384;
+    private static final int DOUBLE = -385;
+    private static final int BYTE = -386;
+    private static final int TEXT = -387;
+
+    private WritableSerializerHooks() {
+    }
+
+    static final class Boolean extends WritableSerializerHook<BooleanWritable> {
+        Boolean() {
+            super(BooleanWritable.class, BooleanWritable::new, BOOLEAN);
+        }
+    }
+
+    static final class Byte extends WritableSerializerHook<ByteWritable> {
+        Byte() {
+            super(ByteWritable.class, ByteWritable::new, BYTE);
+        }
+    }
+
+    static final class Double extends WritableSerializerHook<DoubleWritable> {
+        Double() {
+            super(DoubleWritable.class, DoubleWritable::new, DOUBLE);
+        }
+    }
+
+    static final class Float extends WritableSerializerHook<FloatWritable> {
+        Float() {
+            super(FloatWritable.class, FloatWritable::new, FLOAT);
+        }
+    }
+
+    static final class Int extends WritableSerializerHook<IntWritable> {
+        Int() {
+            super(IntWritable.class, IntWritable::new, INT);
+        }
+    }
+
+    static final class Long extends WritableSerializerHook<LongWritable> {
+        Long() {
+            super(LongWritable.class, LongWritable::new, LONG);
+        }
+    }
+
+    static final class Text extends WritableSerializerHook<org.apache.hadoop.io.Text> {
+        Text() {
+            super(org.apache.hadoop.io.Text.class, org.apache.hadoop.io.Text::new, TEXT);
+        }
+    }
+
+    static final class Default implements SerializerHook<Writable> {
+
+        @Override
+        public Class<Writable> getSerializationType() {
+            return Writable.class;
+        }
+
+        @Override
+        public Serializer createSerializer() {
+            return new WritableStreamSerializer();
+        }
+
+        @Override
+        public boolean isOverwritable() {
+            return true;
+        }
+    }
+
+    private static class WritableStreamSerializer implements StreamSerializer<Writable> {
+
+        private static final ILogger LOGGER = Logger.getLogger(WritableStreamSerializer.class);
+
+        private boolean warnedOnce;
```

Comment:
Yes, I didn't want to do a volatile read for every time this method is called. It's not so important if there's more than one warning, as long as it doesn't flood the logs (which would happen if it warned everytime)